kuzzle-sdk-java/src/main/java/io/kuzzle/sdk/CoreClasses/Maps/CustomMap.java

```diff
@@ -0,0 +1,217 @@
+ * manipulate them.
+ */
+public class CustomMap extends ConcurrentHashMap<String, Object> {
+
+    /**
+     * Convert à ConcurrentHashMap<String, Object> to a CustomMap
+     * @param map ConcurrentHashMap<String, Object> representing JSON.
+     * @return a CustomMap instance
+     */
+    public static CustomMap getCustomMap(ConcurrentHashMap<String, Object> map) {
+        if (map instanceof CustomMap) {
+            return (CustomMap)map;
+        }
+        return new CustomMap(map);
+    }
+
+    /**
+     * Create a new instance of CustomMap
+     */
+    public CustomMap() {
+        super();
+    }
+
+    /**
+     * Create a new instance of CustomMap from a ConcurrentHashMap<String, Object>.
+     * @param map ConcurrentHashMap<String, Object> representing JSON.
+     */
+    public CustomMap(ConcurrentHashMap<String, Object> map) {
+        super(map);
+    }
+
+    /**
+     * Avoid trying to put null element that would trigger an NullPointerException
+     * @param key a String representing the key.
+     */
+    @Override
+    public Object put(String key, Object value) {
+        if (value != null) {
+            return super.put(key, value);
+        }
+        return null;
+    }
+
+    /**
+     * Check whether the key value is null or not.
+     * @param key a String representing the key.
+     * @return true if the key is null.
+     */
+    public boolean isNull(String key) {
+        return super.get(key) == null;
+    }
+    /**
+     * Check whether the key value is a String or not.
+     * @param key a String representing the key.
+     * @return true if the key is a String.
+     */
+    public boolean isString(String key) {
+        return super.get(key) instanceof String;
+    }
+
+    /**
+     * Check whether the key value is a Boolean or not.
+     * @param key a String representing the key.
+     * @return true if the key is a Boolean.
+     */
+    public boolean isBoolean(String key) {
+        return super.get(key) instanceof Boolean;
+    }
+
+    /**
+     * Check whether the key value is a Number or not.
+     * @param key a String representing the key.
+     * @return true if the key is a Number.
+     */
+    public boolean isNumber(String key) {
+        return super.get(key) instanceof Number;
+    }
+
+    /**
+     * Check whether the key value is an ArrayList or not.
+     * @param key a String representing the key.
+     * @return true if the key is an ArrayList.
+     */
+    public boolean isArrayList(String key) {
+        return super.get(key) instanceof ArrayList;
+    }
+
+    /**
+     * Check whether the key value is a ConcurrentHashMap or not.
+     * @param key a String representing the key.
+     * @return true if the key is a ConcurrentHashMap.
+     */
+    public boolean isMap(String key) {
+        return super.get(key) instanceof ConcurrentHashMap;
+    }
+
+    /**
+     * Return the specified key value or null if the value is not a String.
```

Comment:
the Object `ConcurrentHashMap` doesn't throw when you do a `get` on a key that doesn't exists, instead they just return null, I'm also applying this logic, if key does not exists or is not of the good type you just receive a `null` value, this will avoid doing a massive amount of `try catch` to know if you got the value you expected when manipulating JSON data.